services/parse_pairs.go

```diff
@@ -0,0 +1,117 @@
+package services
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/beyondstorage/go-storage/v4/pairs"
+	. "github.com/beyondstorage/go-storage/v4/types"
+)
+
+var (
+	// ErrConnectionStringInvalid means the connection string is invalid.
+	ErrConnectionStringInvalid = NewErrorCode("connection string is invalid")
+)
+
+// <type>://[<name>][<work_dir>][?key1=value1&...&keyN=valueN]
+func parseString(ConnStr string) (ty string, ps []Pair, err error) {
```

Comment:
`parseString` looks too general to me, how about using `parseConnectionString`?

Reply:
I agree this is not a good name, but does `ConnectionString` seem too long (I'm ok with it though)?

Reply:
Long but clear, I think it's OK.